packages/commonjs/test/test.js

```diff
@@ -665,3 +665,22 @@ test('imports .cjs file extension by default', async (t) => {
   const code = await getCodeFromBundle(bundle);
   t.snapshot(code);
 });
+
+test('does not wrap commonjsRegister calls in createCommonjsModule', async (t) => {
+  const bundle = await rollup({
+    input: 'fixtures/samples/dynamic-require-double-wrap/main.js',
+    plugins: [commonjs({
+      sourceMap: true,
+      dynamicRequireTargets: [
+        'fixtures/samples/dynamic-require-double-wrap/submodule.js'
+      ]
+    })]
+  });
+
+  const code = await getCodeFromBundle(bundle, { exports: 'named' });
```

Comment:
If you run this test but add a parameter `format: "es"` here and console.log what the code actually is, you will see that it is invalid as tries to assign something to a non-existing (in an ES module) `module.exports`, which would throw at runtime.

Not sure we can keep it as it is. The only way to fix this is to get rid of all exports, which seems rather complex for me considering it is not entirely sure how this feature will evolve.

Reply:
I logged the test result, and it does not seem to generate anything invalid.
It's assigning to `module.exports` while `module` is valid and passed as an argument from `commonjsRegister`'s callback.

Reply:
Ah, sorry, did not read your reply properly, will need to have another look again myself

Reply:
Oh you are right and I stand corrected. In that case there is no issue from my side.